lib/nosedrum/storage.ex

```diff
@@ -92,13 +92,27 @@ defmodule Nosedrum.Storage do
   ## Return value
   Returns `:ok` if successful, and `{:error, reason}` otherwise.
   """
-  @callback add_command(
+  @callback queue_command(
               name_or_path :: String.t() | application_command_path,
               command_module :: module,
-              scope :: command_scope,
               name_or_pid
             ) :: :ok | {:error, Nostrum.Error.ApiError.t()}
 
+  @doc """
+  Add a new command under the given name or application command path.
+
+  If the command already exists, it will be overwritten.
```

Comment:
Should we document here that when many commands are being registered, it is recommended to use `queue_command` and `process_queue_commands` instead?